### PR TITLE
Implementation of the folder creation feature

### PR DESCRIPTION
**Pull request by Oleh-Stefanovych**
Now there is a `Create Folder` button on the files page for creating a new folder.

The `ESLint` result:
image

### CODE REVIEW

**Comment by Oleh-Stefanovych** [comment]
@dkalinin, PTAL.

File: app/store/actions.js [hunk 1]
```diff
@@ -106,9 +108,61 @@ export default {
...
+        return Promise.reject(new Error('New folder is already creating.'));
+      }
+
+      const baseName = 'New Folder ';
+      let folderNumber = 1;
+
+      while (store.state.files.some((file) => file.name.toLowerCase() ===
+        (baseName + folderNumber).toLowerCase())) {
+        folderNumber++;
+      }
```
Comment (dkalinin) [inline app/store/actions.js]: Let's use .forEach().

File: app/store/actions.js [hunk 2]
```diff
@@ -106,9 +108,61 @@ export default {
...
+      const newFiles = store.state.files.filter((file) => {
+        return file.id !== undefined;
+      });
+      if (newFiles.length !== store.state.files.length) {
+        store.commit('setFiles', newFiles);
+      }
+    };
+  },
+
+  createNewFolder(store) {
```
Comment (dkalinin) [inline app/store/actions.js]: Let's rename to `completeFolderCreation`.

File: app/store/actions.js [hunk 3]
```diff
@@ -106,9 +108,61 @@ export default {
...
+    return (folderName) => {
+      store.commit('setFileListStatus', 'Creating folder...');
+      const newFolder = store.state.files.find((file) => file.id === undefined);
+      newFolder.name = folderName;
+      return new Promise((resolve, reject) => {
```
Comment (dkalinin) [inline app/store/actions.js]: This is not needed.

File: app/store/actions.js (hunk 1 above)
Comment (Oleh-Stefanovych) [reply app/store/actions.js]: Let's discuss in person.

**Comment by Oleh-Stefanovych** [comment]
@dkalinin, PTAL.

**Pull request merged**